addon/db-collection.js

```diff
@@ -29,32 +33,25 @@ class DbCollection {
 
   insert(data) {
     let copy = data ? stringify(data) : {};
-    let records = this._records;
-    let returnData;
 
     if (!_isArray(copy)) {
-      let attrs = copy;
-      if (attrs.id === undefined || attrs.id === null) {
-        attrs.id = records.length + 1;
-      }
-
-      records.push(attrs);
-      returnData = stringify(attrs);
-
+      return this._insertRecord(copy);
     } else {
-      returnData = [];
-      copy.forEach(data => {
-        if (data.id === undefined || data.id === null) {
-          data.id = records.length + 1;
-        }
-
-        records.push(data);
-        returnData.push(data);
-        returnData = returnData.map( r => stringify(r) );
-      });
+      // Need to sort in order to ensure IDs inserted in the correct order
+      return copy
+        .sort(function(a, b) {
```

Comment:
This kind of makes me feel that the implementation is dancing around a weird edge case. Maybe the implementation should tracked used IDs instead of just being a counter?